File: Chapter03/FS/VariableFormText/VariableFormattedText/VariableFormattedText/VariableFormattedTextPage.fs
namespace VariableFormattedText

open Xamarin.Forms

type VariableFormattedTextPage() = 
    inherit ContentPage()
    let formattedString = FormattedString()
    let largeSize = Device.GetNamedSize(NamedSize.Large, typeof<Label>)

    do formattedString.Spans.Add(Span(Text = "I "))
    do formattedString.Spans.Add(Span(Text = "love",
                                      FontSize = largeSize,
                                      FontAttributes = FontAttributes.Bold))
    do formattedString.Spans.Add(Span(Text = " Xamarin.Forms")) 

    do base.Content <- Label(FormattedText = formattedString,
                             VerticalOptions = LayoutOptions.Center,
                             HorizontalOptions = LayoutOptions.Center,
                             FontSize = largeSize)


    

File: Chapter03/VarFormText/VariableFormattedText/VariableFormattedText/VariableFormattedTextPage.cs
using System;
using Xamarin.Forms;

namespace VariableFormattedText
{
    public class VariableFormattedTextPage : ContentPage
    {
        public VariableFormattedTextPage()
        {
            Content = new Label
            {
                FormattedText = new FormattedString
                {
                    Spans =
                    {
                        new Span
                        {
                            Text = "I "
                        },
                        new Span
                        {
                            Text = "love",
                            FontSize = Device.GetNamedSize(NamedSize.Large, typeof(Label)),
                            FontAttributes = FontAttributes.Bold
                        },
                        new Span
                        {
                            Text = " Xamarin.Forms!"
                        }
                    }
                },

                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions = LayoutOptions.Center,
                FontSize = Device.GetNamedSize(NamedSize.Large, typeof(Label))
            };
        }
    }
}
